### ISSUE TITLE
Perl i686 broken on staging

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#27745]
## Issue description

```
Running pm_to_blib for cpan/parent directly
LD_LIBRARY_PATH=/build/perl-5.24.1 ./miniperl -Ilib make_ext.pl cpan/perlfaq/pm_to_blib  MAKE="make" LIBPERL_A=libperl.so

Running pm_to_blib for cpan/perlfaq directly
LD_LIBRARY_PATH=/build/perl-5.24.1 ./miniperl -Ilib make_ext.pl cpan/podlators/pm_to_blib  MAKE="make" LIBPERL_A=libperl.so
Generating a Unix-style Makefile
Writing Makefile for Pod
make[1]: Entering directory '/build/perl-5.24.1/cpan/podlators'
"../../miniperl" "-I../../lib" "-I../../lib" "-I../../lib" "-I../../lib" scripts/pod2man.PL scripts/pod2man
Extracting pod2man (with variable substitutions)
"../../miniperl" "-I../../lib" "-I../../lib" "-I../../lib" "-I../../lib" scripts/pod2text.PL scripts/pod2text
Extracting pod2text (with variable substitutions)
cp scripts/pod2man blib/script/pod2man
/build/perl-5.24.1/cpan/podlators/../../miniperl "-I../../lib" "-I../../lib" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/pod2man
cp scripts/pod2text blib/script/pod2text
/build/perl-5.24.1/cpan/podlators/../../miniperl "-I../../lib" "-I../../lib" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/pod2text
Can't locate Getopt/Long.pm in @INC (you may need to install the Getopt::Long module) (@INC contains: /build/perl-5.24.1/cpan/AutoLoader/lib /build/perl-5.24.1/dist/Carp/lib /build/perl-5.24.1/dist/PathTools /build/perl-5.24.1/dist/PathTools/lib /build/perl-5.24.1/cpan/ExtUtils-Install/lib /build/perl-5.24.1/cpan/ExtUtils-MakeMaker/lib /build/perl-5.24.1/cpan/ExtUtils-Manifest/lib /build/perl-5.24.1/cpan/File-Path/lib /build/perl-5.24.1/ext/re /build/perl-5.24.1/dist/Term-ReadLine/lib /build/perl-5.24.1/dist/Exporter/lib /build/perl-5.24.1/ext/File-Find/lib /build/perl-5.24.1/cpan/Text-Tabs/lib /build/perl-5.24.1/dist/constant/lib /build/perl-5.24.1/cpan/version/lib /build/perl-5.24.1/lib ../../lib .) at /build/perl-5.24.1/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Command/MM.pm line 109.
Makefile:445: recipe for target 'manifypods' failed
make[1]: *** [manifypods] Error 2
make[1]: Leaving directory '/build/perl-5.24.1/cpan/podlators'
make[1]: Entering directory '/build/perl-5.24.1/cpan/podlators'
Can't locate Getopt/Long.pm in @INC (you may need to install the Getopt::Long module) (@INC contains: /build/perl-5.24.1/cpan/AutoLoader/lib /build/perl-5.24.1/dist/Carp/lib /build/perl-5.24.1/dist/PathTools /build/perl-5.24.1/dist/PathTools/lib /build/perl-5.24.1/cpan/ExtUtils-Install/lib /build/perl-5.24.1/cpan/ExtUtils-MakeMaker/lib /build/perl-5.24.1/cpan/ExtUtils-Manifest/lib /build/perl-5.24.1/cpan/File-Path/lib /build/perl-5.24.1/ext/re /build/perl-5.24.1/dist/Term-ReadLine/lib /build/perl-5.24.1/dist/Exporter/lib /build/perl-5.24.1/ext/File-Find/lib /build/perl-5.24.1/cpan/Text-Tabs/lib /build/perl-5.24.1/dist/constant/lib /build/perl-5.24.1/cpan/version/lib /build/perl-5.24.1/lib ../../lib .) at /build/perl-5.24.1/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Command/MM.pm line 109.
Makefile:445: recipe for target 'manifypods' failed
make[1]: *** [manifypods] Error 2
make[1]: Leaving directory '/build/perl-5.24.1/cpan/podlators'
Unsuccessful make(cpan/podlators): code=512 at make_ext.pl line 569.
makefile:605: recipe for target 'cpan/podlators/pm_to_blib' failed
make: *** [cpan/podlators/pm_to_blib] Error 25
builder for ‘/nix/store/fqivvlghbdvafm6cysx1x3i0081gbldf-perl-5.24.1.drv’ failed with exit code 2
```

## Technical details

- Nixpkgs staging at 9be40841ea4251a09ae7a1023abc1b6d191f61bb

**Comment by FRidh** [comment]
It did build before 8ce4780eff33591ed54b2fb97ca96108e4e7a740.

**Comment by globin** [comment]
cc @Ericson2314

**Comment by dezgeg** [comment]
IIRC that (`Can't locate Getopt/Long.pm in @INC`) is a random parallel build issue or something.

**Comment by FRidh** [comment]
It looks indeed familiar. I was able to build it locally so closing.

**Comment by dandedrick** [comment]
For anyone who might stumble across this looking for answers I thought I would share what I found here. This indeed was a parallel build issue with perl. I hit this issue and tracked it down to a missing dependency/race. I've submitted a patch to perl that should fix this issue here: https://rt.perl.org/Public/Bug/Display.html?id=132360

I wasn't actually doing anything with nixos but I thought I would bring some closure to this since it was the only case I could find of someone else running into this issue.
